### PR TITLE
WIP: Feature/transfer

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

### Added
- new settings page for import/export
- new functionality for import/export

### Changed
- Clear cache settings page new UI layout

# Screenshots / Videos

Screenshot 2022-11-22 at 10 28 15
Screenshot 2022-11-23 at 11 11 52

### CODE REVIEW

File: src/Blocks/components/form/assets-admin/transfer.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		const items = button.getAttribute('data-items');
+
+		let output = items ? items.split(",") : [];
+
+		const {value} = element;
+
+		if (element.checked) {
+			output.push(value);
+		} else {
+			output = output.filter((item) => item !== value)
```
Comment (piqusy) [inline src/Blocks/components/form/assets-admin/transfer.js:108]:
```suggestion
			output = output.filter((item) => item !== value);
```

File: src/Blocks/components/form/assets-admin/transfer.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+	createFile(data, exportName) {
+		const dataStr = "data:text/json;charset=utf-8," + encodeURIComponent(data);
+
+		const downloadAnchorNode = document.createElement('a');
+
+		downloadAnchorNode.setAttribute("href", dataStr);
+		downloadAnchorNode.setAttribute("download", exportName + ".json");
+
+		document.body.appendChild(downloadAnchorNode); // required for firefox
```
Comment (piqusy) [inline src/Blocks/components/form/assets-admin/transfer.js:124]:
```suggestion
		document.body.appendChild(downloadAnchorNode); // required for Firefox browser
```

File: src/Blocks/components/form/assets-admin/utilities.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ */
+export const hideGlobalMsg = (selector) => {
+	const messageContainer = document.querySelector(selector);
+
+	if (!messageContainer) {
+		return;
+	}
+
+	messageContainer.classList.remove(FORM_SELECTORS.CLASS_ACTIVE);
+}
```
Comment (piqusy) [inline src/Blocks/components/form/assets-admin/utilities.js:44]:
```suggestion
};
```

File: src/Migration/SettingsMigration.php [hunk 4]
```diff
@@ -80,19 +80,25 @@ public function getSettingsGlobalData(): array
...
-				'submitAttrs' => [
-					'data-type' => self::VERSION_2_3,
-				],
-				'additionalClass' => $manifestForm['componentMigrationJsClass'] . ' es-submit--migration',
+				'component' => 'layout',
+				'layoutItems' => [
+					[
+						'component' => 'card',
+						'cardTitle' => \__('Version 2 to 3'),
+						'cardSubTitle' => \__('In this vesion we have changed the option and custom meta name for the fallback emails. This migration will remove your old options and custom meta keys and updated them to the new ones.', 'eightshift-forms'),
```
Comment (piqusy) [inline src/Migration/SettingsMigration.php:88]:
```suggestion
						'cardSubTitle' => \__('In this version we have changed the option and custom meta name for the fallback emails. This migration will remove your old options and custom meta keys and update them to the new ones.', 'eightshift-forms'),
```

File: src/Transfer/SettingsTransfer.php [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+							],
+						],
+					],
+					[
+						'component' => 'tab',
+						'tabLabel' => \__('Export forms', 'eightshift-forms'),
+						'tabContent' => [
+							[
+								'component' => 'intro',
+								'introSubtitle' => \__('Export one or many forms with settings', 'eightshift-forms'),
```
Comment (piqusy) [inline src/Transfer/SettingsTransfer.php:139]:
```suggestion
								'introSubtitle' => \__('Export single or multiple forms with settings', 'eightshift-forms'),
```

File: src/Transfer/SettingsTransfer.php [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+							],
+						],
+					],
+					[
+						'component' => 'tab',
+						'tabLabel' => \__('Export all', 'eightshift-forms'),
+						'tabContent' => [
+							[
+								'component' => 'intro',
+								'introSubtitle' => \__('Export all form and global settings.', 'eightshift-forms'),
```
Comment (piqusy) [inline src/Transfer/SettingsTransfer.php:162]:
```suggestion
								'introSubtitle' => \__('Export all forms and global settings.', 'eightshift-forms'),
```

**Review by piqusy — APPROVED** [review]
Awesome job! 🚀 
Left some minor spelling/syntax suggestions and linter fixes

File: src/Blocks/components/form/assets-admin/transfer.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+		if (element.checked) {
+			output.push(value);
+		} else {
+			output = output.filter((item) => item !== value)
+		}
+
+		button.disabled = !output.length;
+
+		button.setAttribute('data-items', output);
```
Comment (mbmjertan) [inline src/Blocks/components/form/assets-admin/transfer.js:113]: I think all browser support dataset setters as a bit of syntactic sugar:
```suggestion
		button.dataset.items = output;
```

Comment (iruzevic) [reply src/Blocks/components/form/assets-admin/transfer.js:113]: I like using using function for this so I will leave it like this.

**Pull request merged**